rust/segment/src/types.rs

```diff
@@ -19,6 +19,18 @@ use super::blockfile_record::{
 };
 use super::distributed_hnsw::DistributedHNSWSegmentWriter;
 
+pub(super) fn construct_prefix_path(
+    tenant: &str,
+    database_id: &DatabaseUuid,
+    collection_id: &CollectionUuid,
+    segment_id: &SegmentUuid,
+) -> String {
+    format!(
+        "{}/{}/{}/{}",
```

Comment:
this is different than the prefix used for version files and lineage files (`{}/databases/{}/collections/{}`), I think we should use the same prefix across everything